Mathlib/Analysis/Normed/Ring/Lemmas.lean

```diff
@@ -3,21 +3,22 @@ Copyright (c) 2018 Patrick Massot. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Patrick Massot, Johannes Hölzl
 -/
+import Mathlib.Algebra.Group.AddChar
+import Mathlib.Algebra.Group.TypeTags.Finite
 import Mathlib.Algebra.Order.GroupWithZero.Finset
 import Mathlib.Analysis.Normed.Group.Bounded
 import Mathlib.Analysis.Normed.Group.Int
 import Mathlib.Analysis.Normed.Group.Uniform
 import Mathlib.Analysis.Normed.Ring.Basic
+import Mathlib.GroupTheory.OrderOfElement
+import Mathlib.Topology.MetricSpace.Dilation
 
 /-!
 # Normed rings
 
 In this file we continue building the theory of (semi)normed rings.
 -/
 
--- Guard against import creep.
-assert_not_exists RestrictScalars
```

Comment:
I'm not entirely sure what this was trying to protect against, but maybe we should replace it with a weaker constraint rather than dropping it altogether. Are you able to track down who added it?

Reply:
Oh, yes, I should have explained that. The current files `Normed/Ring/Lemmas` and `Normed/Field/Lemmas` were formerly one file which I split into two halves, and this anti-import constraint got copied into both (my mistake). Since `Normed/Field/Lemmas` imports `Normed/Ring/Lemmas`, it suffices to have it in the former.